posthog/models/session_recording_metadata/session_recording_metadata.py

```diff
@@ -0,0 +1,83 @@
+from django.conf import settings
+
+from ee.kafka_client.topics import KAFKA_SESSION_RECORDING_METADATA
+from posthog.clickhouse.kafka_engine import KAFKA_COLUMNS, kafka_engine
+from posthog.clickhouse.table_engines import Distributed, ReplacingMergeTree, ReplicationScheme
+
+SESSION_RECORDING_METADATA_DATA_TABLE = "sharded_session_recording_metadata"
+
+SESSION_RECORDING_METADATA_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    team_id Int64,
+    distinct_id VARCHAR,
+    session_id VARCHAR,
+    window_id VARCHAR,
+    session_start DateTime64(6, 'UTC'),
+    session_end DateTime64(6, 'UTC'),
+    duration Int64,
+    snapshot_data_location VARCHAR
+    {extra_fields}
+) ENGINE = {engine}
+"""
+
+SESSION_RECORDING_METADATA_DATA_TABLE_ENGINE = lambda: ReplacingMergeTree(
+    "session_recording_metadata", ver="_timestamp", replication_scheme=ReplicationScheme.SHARDED
+)
+SESSION_RECORDING_METADATA_TABLE_SQL = lambda: (
+    SESSION_RECORDING_METADATA_TABLE_BASE_SQL
+    + """PARTITION BY toYYYYMMDD(session_end)
+ORDER BY (team_id, session_id, session_start, session_end)
```

Comment:
This sort key doesn't work for listing sessions - all queries basically devolve into full table scans since session_id is unique per session. Having session_start/session_end after that in the sort key doesn't affect behavior at all.

Can you list out what the full list and fetch queries look like against this table? We should design the sort key accordingly.

Reply:
Suggestion for partition and sort key based on these queries:

```
PARTITION BY toYYYYMMDD(session_start)
ORDER BY (team_id, toHour(session_start), session_id)
```

Why this ordering? It's optimized for the list query - we only look at the hours of data where a session started.

I also include session_id for the single session fetching.